chopper_generator/lib/src/generator.dart

```diff
@@ -298,33 +299,44 @@ final class ChopperGenerator
       bool hasBody = body.isNotEmpty || fields.isNotEmpty;
       if (hasBody) {
         if (body.isNotEmpty) {
+          final DartType bodyType = m.parameters
+              .firstWhere((p) => _typeChecker(chopper.Body).hasAnnotationOf(p))
+              .type;
+          final mapExpression = (formUrlEncoded &&
+                  _isMap(bodyType) &&
+                  !_isMapStringString(bodyType))
+              ? _generateMapToStringExpression(refer(body.keys.first))
+              : refer(body.keys.first);
           blocks.add(
-            declareFinal(Vars.body.toString())
-                .assign(refer(body.keys.first))
-                .statement,
+            declareFinal(Vars.body.toString()).assign(mapExpression).statement,
           );
         } else {
           blocks.add(
             declareFinal(Vars.body.toString())
-                .assign(_generateMap(fields))
+                .assign(_generateMap(fields, isStringValue: formUrlEncoded))
                 .statement,
           );
         }
       }
 
       final bool hasFieldMap = fieldMap.isNotEmpty;
       if (hasFieldMap) {
+        final DartType fieldMapType = m.parameters
+            .firstWhere(
+                (p) => _typeChecker(chopper.FieldMap).hasAnnotationOf(p))
+            .type;
+        final map = (formUrlEncoded && !_isMapStringString(fieldMapType))
+            ? _generateMapToStringExpression(refer(fieldMap.keys.first))
+            : refer(fieldMap.keys.first);
```

Comment:
Question: What if you add multiple `@FieldMap()` annotation in one call? Is that something that's possible? Or should we throw a warning?